lib/std/zig/system.zig

```diff
@@ -1294,6 +1318,31 @@ fn elfInt(is_64: bool, need_bswap: bool, int_32: anytype, int_64: anytype) @Type
     }
 }
 
+fn detectAndroidApiLevel() !u32 {
+    var alloc_buf: [384]u8 = undefined;
+    var fba = std.heap.FixedBufferAllocator.init(&alloc_buf);
+    var child = std.process.Child.init(&.{ "/system/bin/getprop", "ro.build.version.sdk" }, fba.allocator());
+    // pass empty EnvMap, no allocator and deinit() required
+    child.env_map = &std.process.EnvMap.init(undefined);
+    child.stdin_behavior = .Ignore;
+    child.stdout_behavior = .Pipe;
+    child.stderr_behavior = .Ignore;
+
+    try child.spawn();
+    errdefer _ = child.kill() catch {};
+
+    // PROP_VALUE_MAX is 92, output is value + newline.
+    // Currently API levels are two-digit numbers, but we want to make sure we never read a partial value.
+    var stdout_buf: [92 + 1]u8 = undefined;
+    var reader = child.stdout.?.readerStreaming(&.{});
+    const n = try reader.interface.readSliceShort(&stdout_buf);
+    const api_level = std.fmt.parseInt(u32, stdout_buf[0 .. n - 1], 10) catch return error.GetpropOutputMalformed;
+
+    const term = try child.wait();
+    if (term != .Exited or term.Exited != 0) return error.GetpropTerminatedAbnormally;
```

Comment:
To clarify, I suggested returning these specific errors because they seemed reasonable to bubble up as part of `DetectError`. But I think maybe you're right that that's leaking implementation details.

So, on second thought, let's log these error conditions as warnings as you did before. They indicate that an important contract of `getprop` and/or `ro.build.version.sdk` has been broken, and we'll definitely want to know about that. We should still ultimately return `ApiLevelQueryFailed` to the caller of `resolveTargetQuery` though.

Reply:
Well, the `errdefer` path could exclude the errors that have more specific logging.